### ISSUE TITLE
False positive on SA1119 for 'with' expression 

### ISSUE THREAD

**Issue by jnm2** [DotNetAnalyzers/StyleCopAnalyzers#3239]
Exactly the same as #3200 but for C# 9 `with` expressions:

There should be no diagnostic here:

```cs
record Foo(int Value)
{
    public Foo Increment() => this with { Value = Value + 1 };

    public Foo Bar()
    {
        // ⚠ SA1119 Statement should not use unnecessary parenthesis
        //     ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
        return (this with { Value = 42 }).Increment();
    }
}
```

This is what happens if you remove the parens:

```cs
record Foo(int Value)
{
    public Foo Increment() => this with { Value = Value + 1 };

    public Foo Bar()
    {
        // CS1002 ; expected           ↓
        return this with { Value = 42 }.Increment();
    }
}
```
